_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp

```diff
@@ -1011,12 +1011,14 @@ mfxStatus CheckAndFixRoi(MfxVideoParam  const & par, ENCODE_CAPS_HEVC const & ca
         // Elimination of invalid rects
         // Driver requirement
         if (ROI->ROI[i].Left >= ROI->ROI[i].Right || ROI->ROI[i].Top >= ROI->ROI[i].Bottom) {
-            ROI->NumROI--;
-            for (mfxU16 j = i; j < ROI->NumROI; j++)
-                ROI->ROI[j] = ROI->ROI[j + 1];
-            changed++;
-            i--;
-            continue;
+            if (!(ROI->ROI[i].Top == 0 && ROI->ROI[i].Left == 0 && ROI->ROI[i].Right == 0 && ROI->ROI[i].Bottom == 0)) {
```

Comment:
@guoyejun , HEVC encode got a new design.  Please extend the change for it. Seems this is the right place for it: https://github.com/Intel-Media-SDK/MediaSDK/blob/f65d9f2091baa584c4855405084255c1858ad809/_studio/mfx_lib/encode_hw/hevc/agnostic/g9/hevcehw_g9_roi.cpp#L108 .

Reply:
@guoyejun Dmitry meant to introduce your change for both codebases. Also @eistarov could you look at this one?

Reply:
Need to separate ROI check behavior in Query and [run-time](https://github.com/Intel-Media-SDK/MediaSDK/blob/710d7ad7a2f9b4d025b72d5276fcc16be1a01f2e/_studio/mfx_lib/encode_hw/hevc/linux/g9/hevcehw_g9_roi_lin.cpp#L114-L120) - not sure about driver reaction on all-zero ROI.

Reply:
@eistarov  good point, i'm afraid i have no time for such change, just pass the ball to msdk team, thanks.

Reply:
@lakulako , @saosipov , can you please help to finalize the patch?